### DISCUSSION TITLE
Include circuitikz

### DISCUSSION THREAD

**Discussion by yoonghm** [quarto-dev/quarto-cli#6765] [Q&A] [answered]
### Description

It is possible to include circuitikz (https://github.com/circuitikz/circuitikz)[https://github.com/circuitikz/circuitikz] in Quarto? 

Here is an example of electronic circuit:

![image](https://github.com/quarto-dev/quarto-cli/assets/4215991/6c86a600-fd55-408d-9c4d-5be4ca417956)

Here is another example of 3-phase electrical circuit:

![image](https://github.com/quarto-dev/quarto-cli/assets/4215991/94dc690c-91ed-46d1-81d7-c158d526b105)

Mermaid could not draw electrical schematic.

**Comment by mcanouil** [comment, marked as answer]
Since this is based on Tikz and LaTeX, you could simply add that tools in your frontmatter (see <https://quarto.org/docs/output-formats/pdf-basics.html#latex-includes> and write your diagram as raw LaTeX (see <https://quarto.org/docs/output-formats/pdf-basics.html#raw-latex>, but preferably using `` ```{=latex} `` raw block).

Additionally, you might be interested in the following Quarto extension, see <https://github.com/pandoc-ext/diagram>.

**Reply by cderv** [reply]
This `circuitikz` would also make a nice extension for ease of use ! 

@yoonghm How to develop one at: https://quarto.org/docs/extensions/

**Reply by yoonghm** [reply]
I wish to contribute my efforts to Quarto. Are there software design documents (such as functional block diagrams, data flow, etc.) for Quarto? I gather that it calls pandoc with different parameters. It assumes default parameters for page layout, paragraph and character styles. However, pandoc also assumes certains defaults which may be subjected to changed by its contributors. This make the maintenance and enhancement works complex. Could you advise? Thanks.

**Reply by mcanouil** [reply]
I apologise but I don't understand what you are asking here.
Quarto website includes API documentation and documentation to create extension, what else do you need?

**Reply by cscheid** [reply]
He's asking for an internals guide to Quarto (which we sorely need, but we don't have yet, and it's likely to come as we work on a big restructuring in 1.5).
